### PR TITLE
deterministic seed calculation and game params obtaining

### PR DESCRIPTION
**Pull request by algys**

### CODE REVIEW

File: sdk/include/game-contract-sdk/game_base.hpp [hunk 1]
```diff
@@ -320,20 +323,39 @@ class game: public eosio::contract {
...
+    CONTRACT_ACTION(close)
+    void close(uint64_t req_id) {
+        auto session_itr = sessions.require_find(req_id, "session with this req_id not found");
+        eosio::check(is_expired(req_id), "session isn't expired, only expired session can be closed");
+
+        eosio::action(
+            {get_self(),"active"_n},
+            "eosio.token"_n,
+            "transfer"_n,
+            std::make_tuple(get_self(), session_itr->player, session_itr->deposit, std::string("refund"))
```
Comment (justefg) [inline sdk/include/game-contract-sdk/game_base.hpp]: better make transfer  method

```
void transfer(name to, asset quantity) {
    eosio::action(
        eosio::permission_level{_self, "active"_n},
        "eosio.token"_n,
        "transfer"_n,
        std::make_tuple(_self, to, quantity, std::string("player winnings"))
    ).send();
}
```

File: sdk/include/game-contract-sdk/game_base.hpp [hunk 2]
```diff
@@ -221,76 +228,71 @@ class game: public eosio::contract {
...
     void new_game(uint64_t req_id, uint64_t casino_id) {
         const auto session_itr = sessions.require_find(req_id, "session with this req_id not found");
         require_auth({session_itr->player, player_game_permission});
         eosio::check(session_itr->state == static_cast<uint8_t>(state::req_start), "state should be 'req_start'");
         eosio::check(platform::read::is_active_casino(get_platform(), casino_id), "casino is't listed in platform");
+        eosio::check(!is_expired(req_id), "session expired");
+
+        auto casino_addr = platform::read::get_casino(get_platform(), casino_id).contract;
+        casino::game_table casino_games(casino_addr, casino_addr.value);
+        auto game_params = casino_games.get(get_self_id(), "game isn't listed in casino").params;
```
Comment (justefg) [inline sdk/include/game-contract-sdk/game_base.hpp]: const

Comment (algys) [reply sdk/include/game-contract-sdk/game_base.hpp]: done

File: sdk/include/game-contract-sdk/game_base.hpp (hunk 1 above)
Comment (algys) [reply sdk/include/game-contract-sdk/game_base.hpp]: done

**Review by justefg — APPROVED** [review]

**Pull request merged**